### PR TITLE
[Backend][Settings] Settings through UI 

### PR DESCRIPTION
**Pull request by mohitdeuex**
### Describe your changes :

Added Global Settings  and ActivityFeedFilter

#
### Type of change :

- [x] Improvement
- [x] New feature
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/CatalogApplication.java [hunk 1]
```diff
@@ -166,6 +152,30 @@ public void logAfterExecution(StatementContext context) {
     intializeWebsockets(catalogConfig, environment);
   }
 
+  private Jdbi createandSetupJDBI(Environment environment, DataSourceFactory dbFactory) {
```
Comment (harshach) [inline catalog-rest-service/src/main/java/org/openmetadata/catalog/CatalogApplication.java:155]: createAndSetupJDBI

**Comment by vivekratnavel** [comment]
@mohitdeuex Can you please merge the latest main, that should fix any cypress failures

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/CatalogApplicationConfig.java [hunk 2]
```diff
@@ -92,6 +97,39 @@ public class CatalogApplicationConfig extends Configuration {
   @JsonProperty("secretsManagerConfiguration")
   private SecretsManagerConfiguration secretsManagerConfiguration;
 
+  public void fetchActivityFeedConfigurationFromDB(Jdbi jdbi) {
```
[the remark this answers is not in the document]
Comment (mohitdeuex) [reply catalog-rest-service/src/main/java/org/openmetadata/catalog/CatalogApplicationConfig.java]: removed it .. using a scheduled thread to update the activityFeed filter config every 3 mins

[the remark this answers is not in the document]
Comment (mohitdeuex) [reply catalog-rest-service/src/main/java/org/openmetadata/catalog/CatalogApplicationConfig.java]: it's part of Settings now

File: bootstrap/sql/com.mysql.cj.jdbc.Driver/v004__create_db_connection_info.sql [hunk 3]
```diff
@@ -90,3 +90,11 @@ ALTER TABLE thread_entity
 UPDATE dbservice_entity
 SET json = JSON_REMOVE(json, '$.connection.config.databaseSchema','$.connection.config.oracleServiceName')
 WHERE serviceType = 'Oracle';
+
+CREATE TABLE IF NOT EXISTS openmetadata_settings (
+     id MEDIUMINT NOT NULL AUTO_INCREMENT,
+     config_type VARCHAR(36) NOT NULL,
```
Comment (harshach) [inline bootstrap/sql/com.mysql.cj.jdbc.Driver/v004__create_db_connection_info.sql:96]: @mohitdeuex use camelCase. Lets follow the existing style across the code

File: bootstrap/sql/org.postgresql.Driver/v004__create_db_connection_info.sql [hunk 4]
```diff
@@ -84,4 +84,12 @@ ALTER TABLE thread_entity
 
 UPDATE dbservice_entity
 SET json = json::jsonb #- '{connection,config,databaseSchema}' #- '{connection,config,oracleServiceName}'
-WHERE serviceType = 'Oracle';
\ No newline at end of file
+WHERE serviceType = 'Oracle';
+
+CREATE TABLE IF NOT EXISTS openmetadata_settings (
+     id SERIAL NOT NULL ,
+     config_type VARCHAR(36) NOT NULL,
```
Comment (harshach) [inline bootstrap/sql/org.postgresql.Driver/v004__create_db_connection_info.sql:91]: same here

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/CatalogApplication.java [hunk 5]
```diff
@@ -245,7 +253,7 @@ private void registerAuthorizer(CatalogApplicationConfig catalogConfig, Environm
 
   private void registerEventFilter(CatalogApplicationConfig catalogConfig, Environment environment, Jdbi jdbi) {
     if (catalogConfig.getEventHandlerConfiguration() != null) {
-      ContainerResponseFilter eventFilter = new EventFilter(catalogConfig, jdbi);
+      ContainerResponseFilter eventFilter = new ResponseFilter(catalogConfig, jdbi);
```
Comment (harshach) [inline catalog-rest-service/src/main/java/org/openmetadata/catalog/CatalogApplication.java:256]: Lets keep the name as EventFilter

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/CollectionDAO.java [hunk 6]
```diff
@@ -79,10 +95,7 @@
 import org.openmetadata.catalog.type.UsageDetails;
 import org.openmetadata.catalog.type.UsageStats;
 import org.openmetadata.catalog.type.Webhook;
-import org.openmetadata.catalog.util.EntitiesCount;
-import org.openmetadata.catalog.util.EntityUtil;
-import org.openmetadata.catalog.util.FullyQualifiedName;
-import org.openmetadata.catalog.util.ServicesCount;
+import org.openmetadata.catalog.util.*;
```
Comment (harshach) [inline catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/CollectionDAO.java:98]: please avoid wild-card imports

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/CollectionDAO.java [hunk 7]
```diff
@@ -2846,4 +2862,80 @@ interface UtilDAO {
...
+      return getSettings(SettingsType.fromValue(rs.getString("config_type")), rs.getString("json"));
+    }
+
+    public static Settings getSettings(SettingsType configType, String json) {
+      Settings settings = new Settings();
+      settings.setConfigType(configType);
+      Object value = null;
+      try {
+        switch (configType) {
+          case AUTHENTICATION_CONFIGURATION:
```
Comment (harshach) [inline catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/CollectionDAO.java:2878]: lets remove all of these options and keep only the activity feed

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/SettingsRepository.java [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+      return Response.status(500, "Internal Server Error. Reason :" + ex.getMessage()).build();
+    }
+    if (oldValue == null) {
+      return (new RestUtil.PutResponse<>(Response.Status.CREATED, setting, RestUtil.ENTITY_CREATED)).toResponse();
+    } else {
+      return (new RestUtil.PutResponse<>(Response.Status.OK, setting, RestUtil.ENTITY_UPDATED)).toResponse();
+    }
+  }
+
+  public Response updateEntityFilter(String entityType, List<Filters> filters) {
```
Comment (harshach) [inline catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/SettingsRepository.java:66]: You have FilterUtil , why don't we use that to have all the code related to EntityFilter in one place

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/resources/settings/SettingsResource.java [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+      return;
+    }
+    String jsonDataFile = jsonDataFiles.get(0);
+    try {
+      String json = IOUtil.toString(getClass().getClassLoader().getResourceAsStream(jsonDataFile));
+      List<Settings> settings = JsonUtils.readObjects(json, Settings.class);
+      settings.forEach(
+          (setting) -> {
+            try {
+              Settings storedSettings = settingsRepository.getConfigWithKey(setting.getConfigType().toString());
```
Comment (harshach) [inline catalog-rest-service/src/main/java/org/openmetadata/catalog/resources/settings/SettingsResource.java:81]: what happens when there is change after the server start here.

File: catalog-rest-service/src/main/resources/json/data/settings/settingsData.json [hunk 10]
```diff
@@ -0,0 +1,81 @@
+[
+  {
+    "config_type": "airflowConfiguration",
+    "config_value": {
```
Comment (harshach) [inline catalog-rest-service/src/main/resources/json/data/settings/settingsData.json:4]: use camelCase and have only settings for activityFeed

**Review by harshach — CHANGES_REQUESTED** [review]

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/resources/settings/SettingsResource.java (hunk 9 above)
Comment (mohitdeuex) [reply catalog-rest-service/src/main/java/org/openmetadata/catalog/resources/settings/SettingsResource.java:88]: we were using a threaded call to db , which has been changed to use Guava cache like we use for Policy

**Review by harshach — APPROVED** [review]

**Pull request merged**